content/client-lib-development-guide/features.textile

```diff
@@ -81,9 +81,11 @@ h3(#restclient). RestClient
 *** @(RSC7b1)@ When it is not possible to send the @X-Ably-Lib@ header, such as for @JSONP@ requests, the library version should be sent as a query param such as @lib=js-1.0.0@
 ** @(RSC7c)@ If the @addRequestIds@ client option is enabled, every REST request to Ably should include in a @request_id@ query string parameter a random string obtained by url-safe base64-encoding a sequence of at least 9 bytes obtained from a source of randomness. This request ID must remain the same if a request is retried to a fallback host per @RSC15@. Any log messages associated with the request should include the request ID. If the request fails, the request ID must be included in the @ErrorInfo@ returned to the user.
 ** @(RSC7d)@ An @Agent@ library identifier is used to identify the library type and version, as well as relevant information describing any underlying layers and the client platform.
-*** @(RSC7d1)@ The @Agent@ library identifier is composed of a series of @key[/value]@ entries joined by spaces where @key@ is the name of a product and @value@ is an optional version of the associated product which must be a valid "semantic version":https://semver.org/. This must include at least the @<library name>/<library version>@ entry followed optionally, in dependency order, the @name[/version]@ entry for underlying library layer or platform layers. An example would be @ably-flutter/1.2.0 ably-java/1.2.1 android/24@. The documentation for each library must define the content and meaning of the @Agent@ identifiers advertised by that library.
+*** @(RSC7d1)@ The @Agent@ library identifier is composed of a series of @key[/value]@ entries joined by spaces where @key@ is the name of a product and @value@ is an optional version of the associated product. This must include at least the @<library name>/<library version>@ entry plus, optionally, the @name[/version]@ entry for underlying library layer or platform layers. An example would be @ably-flutter/1.2.0 ably-java/1.2.1 android/24@.
 *** @(RSC7d2)@ Where possible, the @Agent@ library identifier should be included in all HTTP and WebSocket requests to the Ably endpoint via an @Ably-Agent@ HTTP header.
 *** @(RSC7d3)@ Where the library is unable to specify a header when making a request (such as for @JSONP@ requests), the @Agent@ library identifier should be sent via an @agent@ query param whose value is the URI-encoded @Agent@ library identifier.
+*** @(RSC7d4)@ Where possible, product versions should be represented as a valid "semantic version":https://semver.org/. This is best-effort as we recognise that strict SemVer is not always available, or otherwise possible to derive from information available at runtime, especially in the case of products that represent operating system versions.
```

Comment:
I don't think "where possible, use semver" is now what we really intend. We're saying "use the version identifier advertised by the component in question". There presumably also should be some rules about how we escape those things so as to not interfere with the parsing of the full agent string, and to ensure that the full string does not contain characters that are not allowed in a header string.

Reply:
This comes down to interpretation of this spec point so if I've made it unclear then I am happy to try to evolve it.

What you capture in your comment here was intended to implicit in what I wrote in the spec point. I am, indeed, saying that the version identifier used by the component in question (a.k.a. product) should be used. But what I am also saying is that if we have any control over the versioning system in use then we should elect to use SemVer (i.e. for our own libraries as well as wrappers we create - upper layers like Flutter plugin).

Reply:
@paddybyers and I have further discussed this [internally](https://ably-real-time.slack.com/archives/C030C5YLY/p1626940588283900?thread_ts=1626839568.280300&cid=C030C5YLY) and it has been agreed that we're both, fundamentally saying the same thing.

As such, and given the number of approvers on this PR already, we'll land this pull request as it stands. This area of the spec can always be subsequently iterated if required, but for the time being it's fit for purpose.